deps/chakrashim/src/v8value.cc

```diff
@@ -45,11 +45,11 @@ static bool IsOfType(const Value* ref, JsValueType type) {
 }
 
 bool Value::IsUndefined() const {
-  return IsOfType(this, JsValueType::JsUndefined);
+    return this == jsrt::GetUndefined();
 }
 
 bool Value::IsNull() const {
-  return IsOfType(this, JsValueType::JsNull);
+    return this == jsrt::GetNull();
```

Comment:
Since we also cache `jsrt::GetTrue()` and `jsrt::GetFalse()` should we also update `Value::IsTrue()` and `Value::IsFalse` to just check for reference equality in the same way? I believe that these would all be equally valid, as long as we continue to have only one script context.

Reply:
It was already being done (long way but done) similarly hence didn't update those. Good idea.